WORKSPACE

```diff
@@ -4,8 +4,11 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "6776d68ebb897625dead17ae510eac3d5f6342367327875210df44dbe2aeeb19",
-    urls = ["https://github.com/bazelbuild/rules_go/releases/download/0.17.1/rules_go-0.17.1.tar.gz"],
+    sha256 = "f04d2373bcaf8aa09bccb08a98a57e721306c8f6043a2a0ee610fd6853dcde3d",
+    urls = [
+        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/rules_go/releases/download/0.18.6/rules_go-0.18.6.tar.gz",
+        "https://github.com/bazelbuild/rules_go/releases/download/0.18.6/rules_go-0.18.6.tar.gz",
```

Comment:
Are there any dependency updates in here that will misalign the bazel build and `go.sum`, dependency versions?

Reply:
I suppose this might already be the case judging by https://github.com/grpc-ecosystem/grpc-gateway/pull/956#issuecomment-503152532. As long as we can open another PR after this try to sync the two, I'm okay with this.

Reply:
There are dependency updates, but it seems it was already misaligned. I don't know how to keep them aligned in future; how important is it? Shouldn't the tests failed if a material difference in dependency versions exists?

Reply:
Is it wishful thinking to think we should be able to keep them in sync? I don't expect us to have enough test coverage to discover a difference in behavior of a single dependency :grimacing:. I won't insist on it, but it would be nice to have.

Reply:
I looked into using Gazelle's update-repos, but it's painful:

1. The definitions it creates are ignored due to https://github.com/bazelbuild/bazel/issues/3908. This can be worked around with a very careful restructuring of WORKSPACE, but this breaks due to:
2. The definitions created by Gazelle don't include the patches required by rules_go, so you can't actually use them - you have to rely on the definitions in rules_go's [repositories.bzl](https://github.com/bazelbuild/rules_go/blob/master/go/private/repositories.bzl), which pins the versions. To keep in sync with this we'd have to change the versions used by vgo, but:
3. vgo doesn't let you copy SHAs into a file like Gopkg.toml. We'd have to manually synthesize vgo's weird not-quite-semver strings (eg v0.0.0-20190311183353-d8887717615a) and run these through `go get` for every dependency in repositories.bzl. I'm not even sure this would work, if the invocations of `go get` would change previously "pinned" dependencies. Even if it did work,
4. The dependencies would get out-of-sync next time go.sum changes.

So I'm inclined to give up...

Reply:
Fair enough, lets ship it :joy: